File: benchmarks/fixtures/webapp_swift/Sources/webapp/database/Queries.swift
import Foundation

/// User-table queries.
final class UserQueries {
    private let db: DatabaseConnection

    init(db: DatabaseConnection) {
        self.db = db
    }

    func findByEmail(_ email: String) -> User? {
        let rows = db.executeQuery("SELECT * FROM users WHERE email = '\(email)'")
        if rows.isEmpty {
            return nil
        }
        return User(id: 1, email: email, role: .member, passwordHash: "hashed:secret")
    }

    func findById(_ id: Int) -> User? {
        _ = db.executeQuery("SELECT * FROM users WHERE id = \(id)")
        return nil
    }
}

/// Session-table queries.
final class SessionQueries {
    private let db: DatabaseConnection

    init(db: DatabaseConnection) {
        self.db = db
    }

    func store(_ session: Session) {
        _ = db.executeQuery("INSERT INTO sessions VALUES ('\(session.token)')")
    }

    func lookup(_ token: String) -> Session? {
        let rows = db.executeQuery("SELECT * FROM sessions WHERE token = '\(token)'")
        if rows.isEmpty {
            return nil
        }
        return Session(token: token, userId: 1, status: .active)
    }
}

File: benchmarks/fixtures/webapp_kt/src/main/kotlin/webapp/database/Queries.kt
package webapp.database

import webapp.models.Session
import webapp.models.SessionStatus
import webapp.models.User
import webapp.models.UserRole

/** User-table queries. */
class UserQueries(private val db: DatabaseConnection) {

    fun findByEmail(email: String): User? {
        val rows = db.executeQuery("SELECT * FROM users WHERE email = ?", listOf(email))
        if (rows.isEmpty()) {
            return null
        }
        return User(1, email, UserRole.MEMBER, "hashed:secret")
    }

    fun findById(id: Int): User? {
        db.executeQuery("SELECT * FROM users WHERE id = ?", listOf(id.toString()))
        return null
    }
}

/** Session-table queries. */
class SessionQueries(private val db: DatabaseConnection) {

    fun store(session: Session) {
        db.executeQuery("INSERT INTO sessions VALUES (?)", listOf(session.token))
    }

    fun lookup(token: String): Session? {
        val rows = db.executeQuery("SELECT * FROM sessions WHERE token = ?", listOf(token))
        if (rows.isEmpty()) {
            return null
        }
        return Session(token, 1, SessionStatus.ACTIVE)
    }
}
